app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,585 @@
+import android.widget.LinearLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.cardview.widget.CardView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintLayout.LayoutParams.MATCH_CONSTRAINT
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.content.ContextCompat
+import androidx.core.widget.ImageViewCompat
+import co.condorlabs.customcomponents.COLLAPSIBLE_CARD_VIEW_END_MARGIN
+import co.condorlabs.customcomponents.COLLAPSIBLE_CARD_VIEW_START_MARGIN
+import co.condorlabs.customcomponents.COLLAPSIBLE_INDICATOR_ROTATION
+import co.condorlabs.customcomponents.COLLAPSIBLE_LINE_SEPARATOR_HEIGHT
+import co.condorlabs.customcomponents.COLLAPSIBLE_LINE_SEPARATOR_START_MARGIN
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_ANIMATION_TIME
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_ICON_HEIGHT
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_ICON_WIDTH
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_INDICATOR_MARGIN_END
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_MARGIN
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_MARGIN_FOOTER_TEXT
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_PADDING
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_SUBTITLE_MARGIN_BOTTOM
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_SUBTITLE_TOP_MARGIN
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_TITLE_MARGIN_TOP
+import co.condorlabs.customcomponents.DEFAULT_COLLAPSIBLE_TITLE_TEXT_SIZE
+import co.condorlabs.customcomponents.DEFAULT_STYLE_ATTR
+import co.condorlabs.customcomponents.DEFAULT_STYLE_RES
+import co.condorlabs.customcomponents.MIN_DEFAULT_PADDING
+import co.condorlabs.customcomponents.ONE_HUNDRED_FLOAT
+import co.condorlabs.customcomponents.OPEN_SANS_SEMI_BOLD
+import co.condorlabs.customcomponents.R
+import co.condorlabs.customcomponents.ZERO
+import co.condorlabs.customcomponents.ZERO_FLOAT
+
+/**
+ * @author Alexis Duque on 2019-05-29.
+ * @company Condor Labs
+ * @email eduque@condorlabs.io
+ */
+class CollapsibleView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = ZERO
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private var collapsibleViewId: Int = NO_ID
+    private var requirementsHeight = ZERO
+    private var elementsAreCollapsed = false
+    private var collapseListener: OnCollapseListener? = null
+    private var textHiddenState: String? = null
+    private var textShowState: String? = null
+    private var iconImageView = AppCompatImageView(context).apply { id = R.id.iconImageId }
+    private var sectionTitleTextView = AppCompatTextView(context).apply { id = R.id.titleId }
```

Comment:
tvSectionTitle, review the guide lines and rename the views based on the readme https://github.com/cebroker/android-evercheck-wallet-2.0

Reply:
done